clang/test/Driver/amdgpu-hip-system-arch.c

```diff
@@ -1,4 +1,3 @@
-// REQUIRES: system-linux
```

Comment:
The compile & test on line 47 is failing on z/OS with the following message:
```
clang: error: cannot determine amdgcn architecture: No AMD GPU detected in the system; consider passing it via '--offload-arch'; environment variable CLANG_TOOLCHAIN_PROGRAM_TIMEOUT specifies the tool timeout (integer secs, <=0 is infinite)
```
I'm not sure if there is a way to fix this (eg. wildcard in the expected output or a compiler option).  If there isn't can you add:
```
UNSUPPORTED: system-zos
```

Reply:
This can't really be a property of zos or any other system. It works for me on systems without AMDGPUs and other OSes. You are seeing the error after the environment variable parse error, so why did this proceed beyond that point?

Reply:
That should have been line 26.  The error is causing the failure.  The env var isn't set for that compilation.